fixed factors

keshavgbpecdelhi opened the pull request:
## Related Issuse

- Issue #223

Closes: #223

#### Describe the changes you've made

The changes are as follows : 
- Modified the function Factors to show all the factors of a given number(s) which was initially showing prime factors which was basically the same code written for HCF finding also.
- Modified the README.md with  the functionality of the code and  for LCM and HCF also.

## Checklist:

<!--
Example how to mark a checkbox:-
- [x] My code follows the code style of this project.
-->

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.

## Screenshots

|        Original         |          Updated           |
| :---------------------: | :------------------------: |
| **original screenshot** | <b>updated screenshot </b> |

#### So, Initially "Factors" does the same thing as the HCF "prime factors" does there was no new functionality and not all factors were shown in that. Just like the below Screenshot (here factors and hcf, shows the same result of prime factors only) 
![Screenshot (105)](https://user-images.githubusercontent.com/56075233/111952191-c4a4ca80-8b0a-11eb-846c-663b7a9b7862.png)

### Here is the modifications : 
#### For single input all factors will be shown like the screenshot below :
![Screenshot (102)](https://user-images.githubusercontent.com/56075233/111952301-ec942e00-8b0a-11eb-85bc-e520cdb05489.png)
#### For multiple inputs all factors of individual input will be shown like this screenshot below :
![Screenshot (104)](https://user-images.githubusercontent.com/56075233/111952280-e2722f80-8b0a-11eb-9b45-6ec29503ae94.png)

sairish2001 reviewed (changes requested):
It seems you changed the functionality of Factors button
Don't remove the previous logic for Factors as it is Calculating Prime Factors

Just change Text of factors button to Prime Factors

and

add a new button named factors which will calculate factors with your logic

keshavgbpecdelhi commented:
> It seems you changed the functionality of Factors button
> Don't remove the previous logic for Factors as it is Calculating Prime Factors
> 
> Just change Text of factors button to Prime Factors
> 
> and
> 
> add a new button named factors which will calculate factors with your logic

Factor previously does the same thing which was done by it in hcf so the same code was written twice initially so I think there should not be another factor button but If its not a good idea i will change it.

keshavgbpecdelhi commented:
Can you help, how to create new tag and add functionality to it?

sairish2001 commented:
No no, just create another button, which onclick calls another function

keshavgbpecdelhi commented:
Unable to find the code written for these tags :/
![Screenshot (105)](https://user-images.githubusercontent.com/56075233/112101034-077aa700-8bcc-11eb-9bc5-82d3a1f50f57.png)

khareyash05 commented:
Hey @sairish2001 Should I work on this one?

keshavgbpecdelhi commented:
If I could get help 🥺

khareyash05 commented:
@keshavgbpecdelhi Ya thats what i stated as to help you 
@sairish2001 Can this be a collaborative issue?

khareyash05 commented:
@keshavgbpecdelhi first of all update your repo it isn't updated

sairish2001 commented:
> @keshavgbpecdelhi Ya thats what i stated as to help you
> @sairish2001 Can this be a collaborative issue?

No this can't be collaborative issue

sairish2001 commented:
@keshavgbpecdelhi 
refer to [appjs/logicjavascript.js](https://github.com/sairish2001/makesmatheasy.github.io/blob/main/appjs/logicjavascript.js)

[start](https://github.com/sairish2001/makesmatheasy.github.io/blob/b1e666d69bd7b58ae345751e7103297410689fef/appjs/logicjavascript.js#L389

But actually there's no need to change anything in these functions.
Just create new `function___(){  }` and a new Button in `index.html` then apply onclick to this button

khareyash05 commented:
@keshavgbpecdelhi Update ur forked repo 
use 
```
git checkout main
git fetch upstream(first add git remote add upstream <then this repo's link>
git merge upstream/main
git push origin main
```
This will make merge conflicts go away

keshavgbpecdelhi commented:
Didn't understood what to do, first and then what! => git fetch upstream(first add git remote add upstream <then this repo's link>
I have to write this as it is? => git merge upstream/main

khareyash05 commented:
First of all add upstream using 
``` git remote add upstream https://github.com/sairish2001/MakesMathEasy.git ```
Then do this
```
git checkout main
git fetch upstream
git merge upstream/main
git push origin main
```

keshavgbpecdelhi commented:
This error I got after following the above steps :  

`$ git push origin main
Total 0 (delta 0), reused 0 (delta 0)
To https://github.com/keshavgbpecdelhi/MakesMathEasy.git
 ! [remote rejected] main -> main (refusing to allow a Personal Access Token to create or update workflow `.github/workflows/issue-labeler.yml` without `workflow` scope)
error: failed to push some refs to 'https://github.com/keshavgbpecdelhi/MakesMathEasy.git'`


@khareyash05 pls look into this 🙏🏻

khareyash05 commented:
@keshavgbpecdelhi I am sorry mate I dont know why this error is coming up

keshavgbpecdelhi commented:
@sairish2001 I have done the required changes [@here](https://github.com/sairish2001/MakesMathEasy/pull/333/commits/ad3714badcbac81c29a342f6cdf1010a887327f5) 

If there is a need of anything or if I have to resolve the conflict, please look into this.
Thankyou 🙏🏻

shonali2600 commented:
@keshavgbpecdelhi, go through this https://jessehouwing.net/cannot-push-to-github-without-workflow-scope/, this may help you.

keshavgbpecdelhi commented:
@shonali2600 I read the article, I think this is not related with the current issue we are facing here. Here we are having merge conflicts. If you have something related with that, it would be highly appreciated.
Thanks

keshavgbpecdelhi commented:
Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (main)
$ git status
On branch main
Your branch is ahead of 'origin/main' by 100 commits.
  (use "git push" to publish your local commits)

nothing to commit, working tree clean

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (main)
$ git remote -v
origin  https://github.com/keshavgbpecdelhi/MakesMathEasy.git (fetch)
origin  https://github.com/keshavgbpecdelhi/MakesMathEasy.git (push)
upstream        https://github.com/sairish2001/MakesMathEasy.git (fetch)
upstream        https://github.com/sairish2001/MakesMathEasy.git (push)

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (main)
$ git branch -a
  factors
* main
  remotes/origin/HEAD -> origin/main
  remotes/origin/factors
  remotes/origin/main
  remotes/origin/revert-170-revert-166-issue_93_back
  remotes/upstream/main

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (main)
$ git fetch upstream main
remote: Enumerating objects: 2252, done.
remote: Counting objects: 100% (870/870), done.
remote: Compressing objects: 100% (53/53), done.
remote: Total 2252 (delta 844), reused 823 (delta 817), pack-reused 1382
Receiving objects: 100% (2252/2252), 5.92 MiB | 3.64 MiB/s, done.
Resolving deltas: 100% (1692/1692), completed with 42 local objects.
From https://github.com/sairish2001/MakesMathEasy
 * branch            main       -> FETCH_HEAD
   ba6a93e..9f1b515  main       -> upstream/main

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (main)
$

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (main)
$

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (main)
$ git reset --hard upstream/main
Updating files: 100% (129/129), done.
HEAD is now at 9f1b515 added subtraction of decimal no.s using 10's complement (#1510)

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (main)
$ git checkout factors
Switched to branch 'factors'

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors)
$ git status
On branch factors
nothing to commit, working tree clean

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors)
$ git log
commit ad3714badcbac81c29a342f6cdf1010a887327f5 (HEAD -> factors, origin/factors)
Author: keshavgbpecdelhi <keshavraturi333@gmail.com>
Date:   Thu Mar 25 12:49:04 2021 +0530

    Fixed everything realated to Factors/LCM/HCF in General Maths Section

commit 4f2ba60853aca068c1e4493ef6ba71ac02713218
Author: keshavgbpecdelhi <keshavraturi333@gmail.com>
Date:   Mon Mar 22 12:24:12 2021 +0530

    fixed factors

commit 23ee90a1d755befa48b9bbe47e809ad07f884475 (origin/main, origin/HEAD)
Author: Kanishka Vijayvargiya <kanishkav8jan@gmail.com>
Date:   Thu Mar 18 16:53:58 2021 +0530

    Added warning in matrix calculator when negative value is entered in dimension (#237)

commit 23c0625d58a3846d84e4170dc292b568c2eafd96
Author: Satyam Singh <56315878+satcasm@users.noreply.github.com>
Date:   Thu Mar 18 10:03:10 2021 +0530

    Fixed inline styling (#234)

    * added product-to-sum/sum-to-product identities

    * added complex calculations

    * improved the inline styling and changed pxs to rems

    * improved upon comment lines

    * final changes

    Co-authored-by: Rajinderpal Singh <60106112+sairish2001@users.noreply.github.com>

commit e5644fdbedf9de73513b3e5a2a35c762ca96e439
Author: anjali7786 <57794837+anjali7786@users.noreply.github.com>

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors)
$

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors)
$ git rebase main
First, rewinding head to replay your work on top of it...
Applying: fixed factors
.git/rebase-apply/patch:39: trailing whitespace.
      if(num!=i)
.git/rebase-apply/patch:67: trailing whitespace.
        if(num!=i)
warning: 2 lines add whitespace errors.
Using index info to reconstruct a base tree...
M       README.md
M       appjs/logicjavascript.js
Falling back to patching base and 3-way merge...
Auto-merging appjs/logicjavascript.js
CONFLICT (content): Merge conflict in appjs/logicjavascript.js
Auto-merging README.md
CONFLICT (content): Merge conflict in README.md
error: Failed to merge in the changes.
hint: Use 'git am --show-current-patch' to see the failed patch
Patch failed at 0001 fixed factors
Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort".

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors|REBASE 1/2)
$ code appjs/logicjavascript.js

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors|REBASE 1/2)
$ git rebase --abort

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors)
$ git reset --soft Head~1

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors)
$ git commit --ammend
error: unknown option `ammend'
usage: git commit [<options>] [--] <pathspec>...

    -q, --quiet           suppress summary after successful commit
    -v, --verbose         show diff in commit message template

Commit message options
    -F, --file <file>     read message from file
    --author <author>     override author for commit
    --date <date>         override date for commit
    -m, --message <message>
                          commit message
    -c, --reedit-message <commit>
                          reuse and edit message from specified commit
    -C, --reuse-message <commit>
                          reuse message from specified commit
    --fixup <commit>      use autosquash formatted message to fixup specified commit
    --squash <commit>     use autosquash formatted message to squash specified commit
    --reset-author        the commit is authored by me now (used with -C/-c/--amend)
    -s, --signoff         add Signed-off-by:
    -t, --template <file>
                          use specified template file
    -e, --edit            force edit of commit
    --cleanup <mode>      how to strip spaces and #comments from message
    --status              include status in commit message template
    -S, --gpg-sign[=<key-id>]
                          GPG sign commit

Commit contents options
    -a, --all             commit all changed files
    -i, --include         add specified files to index for commit
    --interactive         interactively add files
    -p, --patch           interactively add changes
    -o, --only            commit only specified files
    -n, --no-verify       bypass pre-commit and commit-msg hooks
    --dry-run             show what would be committed
    --short               show status concisely
    --branch              show branch information
    --ahead-behind        compute full ahead/behind values
    --porcelain           machine-readable output
    --long                show status in long format (default)
    -z, --null            terminate entries with NUL
    --amend               amend previous commit
    --no-post-rewrite     bypass post-rewrite hook
    -u, --untracked-files[=<mode>]
                          show untracked files, optional modes: all, normal, no. (Default: all)


Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors)
$ git commit --amend
[factors 9cb374c] fixed factors
 Date: Mon Mar 22 12:24:12 2021 +0530
 5 files changed, 76 insertions(+), 5 deletions(-)

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors)
$ git log
commit 9cb374ce6cd4ca51bb90935a78edc042cd2c48bb (HEAD -> factors)
Author: keshavgbpecdelhi <keshavraturi333@gmail.com>
Date:   Mon Mar 22 12:24:12 2021 +0530

    fixed factors

commit 23ee90a1d755befa48b9bbe47e809ad07f884475 (origin/main, origin/HEAD)
Author: Kanishka Vijayvargiya <kanishkav8jan@gmail.com>
Date:   Thu Mar 18 16:53:58 2021 +0530

    Added warning in matrix calculator when negative value is entered in dimension (#237)

commit 23c0625d58a3846d84e4170dc292b568c2eafd96
Author: Satyam Singh <56315878+satcasm@users.noreply.github.com>
Date:   Thu Mar 18 10:03:10 2021 +0530

    Fixed inline styling (#234)

    * added product-to-sum/sum-to-product identities

    * added complex calculations

    * improved the inline styling and changed pxs to rems

    * improved upon comment lines

    * final changes

    Co-authored-by: Rajinderpal Singh <60106112+sairish2001@users.noreply.github.com>

commit e5644fdbedf9de73513b3e5a2a35c762ca96e439
Author: anjali7786 <57794837+anjali7786@users.noreply.github.com>
Date:   Wed Mar 17 21:18:47 2021 +0530

    Prime checker & prime till n cal added (#123) (#229)

commit d9dabb64e3c1350d4a623b5233d3cefae5448324
Author: Kanishka Vijayvargiya <kanishkav8jan@gmail.com>

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors)
$

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors)
$ git rebase main
First, rewinding head to replay your work on top of it...
Applying: fixed factors
.git/rebase-apply/patch:42: trailing whitespace.
      if(num!=i)
.git/rebase-apply/patch:70: trailing whitespace.
        if(num!=i)
.git/rebase-apply/patch:95: trailing whitespace.
      if(num!=i)
warning: 3 lines add whitespace errors.
Using index info to reconstruct a base tree...
M       README.md
M       appjs/index.js
M       appjs/logicjavascript.js
M       appjs/searchbar.js
M       index.html
Falling back to patching base and 3-way merge...
Auto-merging index.html
CONFLICT (content): Merge conflict in index.html
Auto-merging appjs/searchbar.js
Auto-merging appjs/logicjavascript.js
CONFLICT (content): Merge conflict in appjs/logicjavascript.js
Auto-merging appjs/index.js
CONFLICT (content): Merge conflict in appjs/index.js
Auto-merging README.md
CONFLICT (content): Merge conflict in README.md
error: Failed to merge in the changes.
hint: Use 'git am --show-current-patch' to see the failed patch
Patch failed at 0001 fixed factors
Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort".

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors|REBASE 1/1)
$ git status
rebase in progress; onto 9f1b515
You are currently rebasing branch 'factors' on '9f1b515'.
  (fix conflicts and then run "git rebase --continue")
  (use "git rebase --skip" to skip this patch)
  (use "git rebase --abort" to check out the original branch)

Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        modified:   appjs/searchbar.js

Unmerged paths:
  (use "git restore --staged <file>..." to unstage)
  (use "git add <file>..." to mark resolution)
        both modified:   README.md
        both modified:   appjs/index.js
        both modified:   appjs/logicjavascript.js
        both modified:   index.html


Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors|REBASE 1/1)
$ code .

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors|REBASE 1/1)
$ git add .

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors|REBASE 1/1)
$ git rebase --continue
Applying: fixed factors

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors)
$ git add .

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors)
$ git rebase --continue
fatal: No rebase in progress?

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors)
$ git push -f
fatal: The current branch factors has no upstream branch.
To push the current branch and set the remote as upstream, use

    git push --set-upstream origin factors


Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors)
$ git push -f origin factors
Enumerating objects: 15, done.
Counting objects: 100% (15/15), done.
Delta compression using up to 4 threads
Compressing objects: 100% (8/8), done.
Writing objects: 100% (8/8), 2.04 KiB | 260.00 KiB/s, done.
Total 8 (delta 7), reused 0 (delta 0)
remote: Resolving deltas: 100% (7/7), completed with 7 local objects.
To https://github.com/keshavgbpecdelhi/MakesMathEasy.git
 ! [remote rejected] factors -> factors (refusing to allow a Personal Access Token to create or update workflow `.github/workflows/Auto_Issue_Closer.yml` without `workflow` scope)
error: failed to push some refs to 'https://github.com/keshavgbpecdelhi/MakesMathEasy.git'

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors)
$ git push -f origin factors
Enumerating objects: 15, done.
Counting objects: 100% (15/15), done.
Delta compression using up to 4 threads
Compressing objects: 100% (8/8), done.
Writing objects: 100% (8/8), 2.04 KiB | 231.00 KiB/s, done.
Total 8 (delta 7), reused 0 (delta 0)
remote: Resolving deltas: 100% (7/7), completed with 7 local objects.
To https://github.com/keshavgbpecdelhi/MakesMathEasy.git
 ! [remote rejected] factors -> factors (refusing to allow a Personal Access Token to create or update workflow `.github/workflows/Auto_Issue_Closer.yml` without `workflow` scope)
error: failed to push some refs to 'https://github.com/keshavgbpecdelhi/MakesMathEasy.git'

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors)
$ git log -1
commit 0daf4bd4164487786ff210b9606c680126086e1e (HEAD -> factors)
Author: keshavgbpecdelhi <keshavraturi333@gmail.com>
Date:   Mon Mar 22 12:24:12 2021 +0530

    fixed factors

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors)
$ git log -2
commit 0daf4bd4164487786ff210b9606c680126086e1e (HEAD -> factors)
Author: keshavgbpecdelhi <keshavraturi333@gmail.com>
Date:   Mon Mar 22 12:24:12 2021 +0530

    fixed factors

commit 9f1b51577955237400ad3a7118649223eeeca1cf (upstream/main, main)
Author: Ansuman Behera <71064674+AB-san@users.noreply.github.com>
Date:   Thu Apr 29 22:25:11 2021 +0530

    added subtraction of decimal no.s using 10's complement (#1510)

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors)
$ git diff head head^
diff --git a/README.md b/README.md
index 21af692..e6922fe 100644
--- a/README.md
+++ b/README.md
@@ -133,12 +133,69 @@ Other Tech/Libraries
     - Upto 20 numbers
   - HCF
     - Calculating Factors of each number and picking common out of them
-  - Prime Factors
-    - Prime Factorization **( with steps for single number)**
   - Factors
-    - All Factores of number(s) including 1 and the number itself
-  - Plot Graph
-    - Plots a Graph of entered Equation having single variable 'x'
+    - Prime Factorization **( with steps for single number)**
+  - Greatest Integer Function and Fractional Part
+  - Solve for Exponents
+  - Sum of Square of Given number
+  - Percentage Calculator
+  - Set Calculator
+  - Pythagorean Triplets
+  - Rank Calculator
+  - Leap Year
+  - Standard Deviation
+  - Hydrostatic Pressure
+  - Strain Calculator
+  - Stress Calculator
+  - Profit Loss Calculations
+  - Statistics Formulas
+  - Mean Calculator
+    - Arithmetic Mean
+    - Geometric Mean
+    - Harmonic Mean
+  - Log Calculator
+    - Calculates the log of x to the base y
+  - Log Properties
+  - Operations on Fractions
+  - Binomial Theorem Properties
+  - Multiplication Table
+    - Prints Multiplication Table
+  - Divide
+    - Division
+    - Divisibility Checker
+  - Multiplication with Steps
+  - Rounding off numbers
+  - EMI Calculator
+  - GST Calculator
+  - Polynomial Degree Finder
+  - Simple and Compound Interest
+  - Factorial
+  - Permutation & Combination
+  - Mean/Median/Mode
+  - Prime Numbers
+    - Primality Test
+    - Primes till entered number
+- Basic Convertors
+  - Currency Convertor
+    - Conversion to words **(eg. '243' to 'two hundred and forty three only')**
+  - Conversion to words
+  - Unit Convertor
+    - Length
+    - Area
+    - Volume
+    - Mass
+    - Temperature
+    - Pressure
+    - Energy
+    - Time
+    - Angle
+    - Speed
+    - Data
+    - Power
+      Improper to Mixed Fraction
+  - SPI/CGPA Convertor
+  - Ascending and Descending Order
+  - Date
   - Roman/Arabic Numerals
     - Conversion from Roman to Arabic and vice versa
     - Expanding the Arabic numbers **(eg. '2423' to '2000+400+20+3')**
diff --git a/appjs/index.js b/appjs/index.js
index 9594b98..0378c3f 100644
--- a/appjs/index.js
+++ b/appjs/index.js
@@ -129,49 +129,185 @@ function collapseit(openit) {
 }

 function openit(id) {
-  var ids = [
-    "#complexcollapse",
-    "#equationssolver",
-    "#mulsolwithsteps",
-    "#table",
-    "#shapescal",
-       "#tdshapescal",
-    "#divide",
-    "#simpletrignocollapse",
-    "#trigonovaluestable",
-    "#trigonoiden",
-    "#factors",
-    "#primefactors",
-    "#integralcollapse",
-    "#differentiatecollapse",
-    "#partialdiffcollapse",
-    "#laplacecollapse",
-    "#limitscollapse",
-    "#matrixcollapse",
-    "#multiplematrixcollapse",
-    "#singlematrixcollapse",
-    "#about",
-    "#rootsquadratic",
-    "#plotgraph",
-    "#roundoff",
-    "#unitconcal",
-    "#home",
-    "#curconcal",
-    "#factorial",
-    "#pandc",
-    "#interest",
-    "#decimal-binary",
-    "#bitwise-calc",
-    "#octal-binary",
-         "#binary-hexadecimal",
-    "#inversetrigonoiden",
-    "#prime",
-  ];
-  for (i = 0; i < ids.length; i++) {
-    if (ids[i] != id) {
-      $(ids[i]).slideUp();
-    } else {
-      $(String(id)).slideToggle();
+    var ids = [
+        "#anyBase",
+        "#complex1collapse",
+        "#complex2collapse",
+        "#complexidentities",
+        "#complexproperties",
+        "#theo",
+        "#impfixs",
+        "#equationssolver",
+        "#mulsolwithsteps",
+        "#secarea",
+        "#table",
+        "#ssts",
+        "#arcs",
+        "#sosqs",
+        "#sosqsn",
+        "#socbsn",
+        "#squaresRanges",
+
+        "#cubesRanges",
+        "#numcubesRanges",
+        "#segcals",
+        "#hydroz",
+        "#straincalc",
+        "#shapescal",
+        "#tdshapescal",
+        "#divide",
+        "#simpletrignocollapse",
+        "#trigonovaluestable",
+        "#trigonoiden",
+        "#factors",
+        "#stats",
+        "#math",
+        "#integralcollapse",
+        "#integration-idencollapse",
+        "#defintegration-idencollapse",
+        "#differentiatecollapse",
+        "#partialdiffcollapse",
+        "#3d-solid",
+        "#laplacecollapse",
+        "#limitscollapse",
+        "#parab",
+        "#matrixcollapse",
+        "#pows",
+        "#matrixprops",
+        "#fractions",
+        "#multiplematrixcollapse",
+        "#singlematrixcollapse",
+        "#algebraic-idencollapse",
+        "#about",
+        "#propcircle",
+        "#propquad",
+        "#pythtriple",
+        "#rootsquadratic",
+        "#plotgraph",
+        "#3dgeocalc",
+        "#plotbargraph",
+        "#roundoff",
+        "#euclid",
+           "#aod",
+           "#vecalg",
+           "#thedif",
+           "#corgeo",
+        "#betagamma",
+        "#parallel",
+        "#unitconcal",
+        "#giff",
+        "#spiconcal",
+        "#home",
+        "#curconcal",
+        "#factorial",
+        "#setop",
+        "#trans",
+        "#hp",
+        "#ppmfs",
+        "#bool",
+        "#eirs",
+        "#log_values",
+        "#log-collapse",
+        "#bt-collapse",
+        "#bt-collapse1",
+        "#pandc",
+        "#stresscalc",
+        "#interest",
+        "#decimal-binary",
+        "#bitwise-calc",
+        "#adding-all",
+        "#subtract-all",
+        "#multiplying-all",
+               "#onetwocom-calc",
+        "#3dgeo",
+        "#bpmfs",
+        "#straightline",
+        "#octal-binary",
+        "#binary-hexadecimal",
+        "#cay",
+        "#octal-hexadecimal",
+        "#inversetrigonoiden",
+        "#hyptrigonoiden",
+        "#invhyptrigonoiden",
+           "#circlecollapse",
+        "#parabolacollapse",
+        "#ellipsecollapse",
+        "#hyperbolacollapse",
+        "#datecal",
+        "#prime",
+        "#sum_n_AP",
+        "#sum_n_GP",
+        "#sum_n_HP",
+        "#algebraic_formulascollapse",
+        "#expansion",
+        "#solutiontri",
+        "#plotangle",
+        "#profitloss",
+        "#differentiate-rulecollapse",
+        "#emical",
+        "#gstcal",
+        "#deviation",
+        "#degcal",
+        "#trigsolcollapse",
+        "#grey-bin",
+        "#consim",
+        "#convdiv",
+        "#curve",
+        "#coor",
+        "#mean",
+        "#Meanit",
+        "#bcd",
+        "#vector",
+        "#vec",
+        "#diffeqn",
+        "#maxmin",
+        "#locroots",
+        "#tangent",
+        "#srf",
+        "#probabilitycollapse",
+        "#joint-probabilitycollapse",
+        "#dec2421",
+        "#ex3",
+        "#lappro",
+        "#rankcal",
+        "#bayes-probabilitycollapse",
+        "#prism",
+        "#pentprism",
+        "#pyramid",
+        "#tripyramid",
+        "#octpyramid",
+        "#hexpyramid",
+        "#octahedron",
+        '#dodecahedron',
+        "#icosahedron",
+        "#anglecon",
+        "#frustum",
+        "#annulus",
+        "#cosine",
+        "#obliquecy",
+        "#setcal",
+        "#mi",
+           "#ip",
+           "#prices",
+        "#centcal",
+        "#cevtha",
+
+        "#percal",
+        "#lyear",
+        "#analytical",
+        "#isoright",
+        "#ellipsoid",
+        "#expo",
+        "#setformula",
+        "#wedge",
+        "#fourier_series"
+    ];
+    for (i = 0; i < ids.length; i++) {
+        if (ids[i] != id) {
+            $(ids[i]).slideUp();
+        } else {
+            $(String(id)).slideToggle();
+        }
     }
 }

@@ -397,3 +533,4 @@ $(document).on('click', ' .list_menu_items ', function () {
     $(this).siblings().removeClass('home');
     $(this).addClass('home');
 });
+
diff --git a/appjs/logicjavascript.js b/appjs/logicjavascript.js
index dfcfd1c..14ce09f 100644
--- a/appjs/logicjavascript.js
+++ b/appjs/logicjavascript.js
@@ -408,99 +408,57 @@ function showdivisionstepsforcheckdivisibility() {
 //factors
 //    one number with steps
 function printfactors() {
-  var num = parseInt(document.getElementById("numforfactorhcflcm").value);
-  var temp = "";
-  for (i = 1; i <= num; i++) {
-    if(num%i==0){
-      temp += i ;
-      if(num!=i)
-       temp += ",";
-    }
-    document.getElementById("factorresult").innerHTML =
-      "\\[All \\space Factors \\space are:\\]\\[" + temp + "\\]";
-    renderMathInElement(document.getElementById("factorresult"));
-  }
-}
-//    one number with steps
-
-//    more than one number without steps
-function printmorefactors(input, output) {
-  document.getElementById(output).textContent = "";
-  var ar = [];
-  var val = document.getElementById(input).value;
-  val = val.replace(/\s+$/, ""); //right trim
-  val = val.replace(/^\s+/, ""); //left trim
-  val = val.split(" ");
-  ar = val;
-  var temp = "";
-  document.getElementById(output).innerHTML =
-    "\\[All \\space Factors \\space of \\space\\]";
-  for (num of ar) {
-    var getnu = "\\[" + num + "\\space : \\space";
-    var i;
-    for (i = 1; i <= num; i++) {
-      if(num%i==0){
-        temp += i ;
-        if(num!=i)
-         temp += ",";
-      }
+    var num = parseInt(document.getElementById("numforfactorhcflcm").value);
+    var temp = "";
+    var tt = "";
+    var v = "";
+    for (i = 2; i <= num; i++) {
+        while (num % i == 0) {
+            temp += i + "&nbsp;&nbsp;&nbsp;<br>";
+            v += i + ",";
+            tt += "<pu>" + num + "</pu><br>";
+            num = num / i;
+        }
+        document.getElementById("dividefactor").innerHTML = temp;
+        document.getElementById("dividefactorresult").innerHTML = tt;
+        document.getElementById("factorresult").innerHTML =
+            "\\[Prime \\space Factors \\space are:\\]\\[" + v.slice(0, -1)+ "\\]";
+        renderMathInElement(document.getElementById("factorresult"));
     }
-    //temp = temp.slice(0, -1);
-    document.getElementById(output).innerHTML += getnu + temp + "\\]";
-    temp += "<br>";
-    temp = "";
-  }
-  renderMathInElement(document.getElementById(output));
-}
-//    more than one number without steps
-
-//Prime factors
-//    one number with steps
-function printprimefactors() {
-  var num = parseInt(document.getElementById("numforfactorhcflcm").value);
-  var temp = "";
-  var tt = "";
-  var v = "";
-  for (i = 2; i <= num; i++) {
-    while (num % i == 0) {
-      temp += i + "&nbsp;&nbsp;&nbsp;<br>";
-      v += i;
-      if(num!=i)
-         v += ",";
-      tt += "<pu>" + num + "</pu><br>";
-      num = num / i;
-    }
-    document.getElementById("dividefactor").innerHTML = temp;
-    document.getElementById("dividefactorresult").innerHTML = tt;
-    //v = v.slice(0, -1);
-    document.getElementById("factorresult").innerHTML =
-      "\\[Prime \\space Factors \\space are:\\]\\[" + v + "\\]";
-    renderMathInElement(document.getElementById("factorresult"));
-  }
 }

 //    one number with steps

 //    more than one number without steps
-function printmoreprimefactors(input, output) {
-  document.getElementById(output).textContent = "";
-  var ar = [];
-  var val = document.getElementById(input).value;
-  val = val.replace(/\s+$/, ""); //right trim
-  val = val.replace(/^\s+/, ""); //left trim
-  val = val.split(" ");
-  ar = val;
-  var temp = "";
-  document.getElementById(output).innerHTML =
-    "\\[Prime \\space Factors \\space of \\space\\]";
-  for (num of ar) {
-    var getnu = "\\[" + num + "\\space : \\space";
-    var i;
-    for (i = 2; i <= num; i++) {
-      while (num % i == 0) {
-        temp += i + ",";
-        num = num / i;
-      }
+function printmorefactors(input, output) {
+    document.getElementById(output).textContent = "";
+    var ar = [];
+    var val = document.getElementById(input).value;
+    val = val.replace(/\s+$/, ""); //right trim
+    val = val.replace(/^\s+/, ""); //left trim
+       if (val.search(/^[0-9 ]+$/) == -1)
+       {
+               document.getElementById("hcfprimefactor").innerHTML ="Enter numbers only";
+               return;
+       }
+    val = val.split(" ");
+    ar = val;
+    var temp = "";
+    document.getElementById(output).innerHTML =
+        "\\[Prime \\space Factors \\space of \\space\\]";
+    for (num of ar) {
+        var getnu = "\\[" + num + "\\space : \\space";
+        var i;
+        for (i = 2; i <= num; i++) {
+            while (num % i == 0) {
+                temp += i + ",";
+                num = num / i;
+            }
+        }
+        temp = temp.slice(0, -1);
+        document.getElementById(output).innerHTML += getnu + temp + "\\]";
+        temp += "<br>";
+        temp = "";
     }
     renderMathInElement(document.getElementById(output));
 }
@@ -570,32 +528,42 @@ function findduplicatesforhcf(array) {
 }

 function hcf(input) {
-  var ar = [];
-  var val = document.getElementById(input).value;
-  val = val.replace(/\s+$/, ""); //right trim
-  val = val.replace(/^\s+/, ""); //left trim
-  val = val.split(" ");
-  ar = val;
-
-  document.getElementById("hcfprimefactor").textContent = "";
-  var temp = "";
-  document.getElementById("hcfprimefactor").innerHTML =
-    "\\[Prime \\space Factors \\space of \\space\\]";
-  var factorarray = [];
-  var j = 0;
-  for (num of ar) {
-    factorarray[j] = [];
-    factorarray[j][0] = 1;
-    var index = 1;
-    var getnu = "\\[" + num + "\\space : \\space";
-    //temp += "1,"; as 1 is not a prime
-    for (i = 2; i <= num; i++) {
-      while (num % i == 0) {
-        factorarray[j][index] = i;
-        index++;
-        temp += i + ",";
-        num = num / i;
-      }
+    var ar = [];
+    var val = document.getElementById(input).value;
+    val = val.replace(/\s+$/, ""); //right trim
+    val = val.replace(/^\s+/, ""); //left trim
+       document.getElementById("hcfprimefactor").textContent = "";
+       if (val.search(/^[0-9 ]+$/) == -1)
+       {
+               document.getElementById("hcfprimefactor").innerHTML ="Enter positive numbers only";
+               return;
+       }
+    val = val.split(" ");
+    ar = val;
+    var temp = "";
+    document.getElementById("hcfprimefactor").innerHTML =
+        "\\[Prime \\space Factors \\space of \\space\\]";
+    var factorarray = [];
+    var j = 0;
+    for (num of ar) {
+        factorarray[j] = [];
+        factorarray[j][0] = 1;
+        var index = 1;
+        var getnu = "\\[" + num + "\\space : \\space";
+        temp += "1,";
+        for (i = 2; i <= num; i++) {
+            while (num % i == 0) {
+                factorarray[j][index] = i;
+                index++;
+                temp += i + ",";
+                num = num / i;
+            }
+        }
+        temp = temp.slice(0, -1);
+        document.getElementById("hcfprimefactor").innerHTML += getnu + temp + "\\]";
+        j++;
+        temp += "<br>";
+        temp = "";
     }

     var arrayoffactors = factorarray;
@@ -661,17 +629,7 @@ function factorselect(numid) {
         printmorefactors(numid, "factorresult");
     }
 }
-function primefactorselect(numid) {
-  var num = document.getElementById(numid).value;
-  num = num.replace(/^\s+/, ""); //left trim
-  num = num.replace(/\s+$/, ""); //right trim
-  num = num.split(" ");
-  if (num.length == 1) {
-    printprimefactors();
-  } else {
-    printmoreprimefactors(numid, "factorresult");
-  }
-}
+
 function printfactorsfl(value) {
     var num = parseInt(value);
     var temp = [];
@@ -8266,5 +8224,3 @@ function ssscal()
     }
     document.getElementById("sstans").innerHTML=ans;
 }
-}
-}
\ No newline at end of file
diff --git a/appjs/searchbar.js b/appjs/searchbar.js
index a587fae..0382e07 100644
--- a/appjs/searchbar.js
+++ b/appjs/searchbar.js
@@ -48,8 +48,6 @@ function autocomplete(inp, arr) {
                         $("#home").slideToggle();
                     } else if (inp.value.toUpperCase() == "FACTORS") {
                         $("#factors").slideToggle();
-                    }else if (inp.value.toUpperCase() == "PRIME FACTORS") {
-                        $("#primefactors").slideToggle();
                     } else if (inp.value.toUpperCase() == "MATRIX") {
                         $("#matrixcollapse").slideToggle();
                     } else if (inp.value.toUpperCase() == "MULTIPLICATION TABLE") {
diff --git a/index.html b/index.html
index e839a1d..b8d9970 100644
--- a/index.html
+++ b/index.html
@@ -1349,36 +1349,6 @@
                 </table>
             </li>

-                    </form>
-                </div>
-                <div class="row">
-                    <p id="resultofvariabledivide" style="font-family:mathfont;;"></p>
-                </div>
-            </div>
-            <!--        start factors-->
-            <div class="collapse" id="factors">
-                <div class="form-group">
-                    <input type="tel" id="numforfactorhcflcm" class="form__field"
-                        placeholder="Enter Number(s) Space Separated">
-                </div>
-                <div class="form-group">
-                    <button type="button" onclick="factorslcmhcfempty();factorselect('numforfactorhcflcm')"
-                        class="btn btn-light">
-                        Factors
-                    </button>
-                    <button type="button" onclick="factorslcmhcfempty();primefactorselect('numforfactorhcflcm')"
-                        class="btn btn-light">
-                        Prime Factors
-                    </button>
-                    <button type="button" onclick="factorslcmhcfempty();lcmsol('numforfactorhcflcm')"
-                        class="btn btn-light">
-                        LCM
-                    </button>
-                    <button type="button" onclick="factorslcmhcfempty();hcf('numforfactorhcflcm')"
-                        class="btn btn-light">
-                        HCF
-                    </button>
-                </div>

             <li>
                 <table class="table">

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors)
$ git push -f
fatal: The current branch factors has no upstream branch.
To push the current branch and set the remote as upstream, use

    git push --set-upstream origin factors


Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors)
$ git push -u -f origin factors
Enumerating objects: 15, done.
Counting objects: 100% (15/15), done.
Delta compression using up to 4 threads
Compressing objects: 100% (8/8), done.
Writing objects: 100% (8/8), 2.04 KiB | 260.00 KiB/s, done.
Total 8 (delta 7), reused 0 (delta 0)
remote: Resolving deltas: 100% (7/7), completed with 7 local objects.
To https://github.com/keshavgbpecdelhi/MakesMathEasy.git
 ! [remote rejected] factors -> factors (refusing to allow a Personal Access Token to create or update workflow `.github/workflows/Auto_Issue_Closer.yml` without `workflow` scope)
error: failed to push some refs to 'https://github.com/keshavgbpecdelhi/MakesMathEasy.git'

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors)
$ git rebase --continue
fatal: No rebase in progress?

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors)
$ git add .

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors)
$ git push -u -f origin factors
Enumerating objects: 15, done.
Counting objects: 100% (15/15), done.
Delta compression using up to 4 threads
Compressing objects: 100% (8/8), done.
Writing objects: 100% (8/8), 2.04 KiB | 231.00 KiB/s, done.
Total 8 (delta 7), reused 0 (delta 0)
remote: Resolving deltas: 100% (7/7), completed with 7 local objects.
To https://github.com/keshavgbpecdelhi/MakesMathEasy.git
 ! [remote rejected] factors -> factors (refusing to allow a Personal Access Token to create or update workflow `.github/workflows/Auto_Issue_Closer.yml` without `workflow` scope)
error: failed to push some refs to 'https://github.com/keshavgbpecdelhi/MakesMathEasy.git'

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors)
$ git push -u -f origin factors
Enumerating objects: 15, done.
Counting objects: 100% (15/15), done.
Delta compression using up to 4 threads
Compressing objects: 100% (8/8), done.
Writing objects: 100% (8/8), 2.04 KiB | 260.00 KiB/s, done.
Total 8 (delta 7), reused 0 (delta 0)
remote: Resolving deltas: 100% (7/7), completed with 7 local objects.
To https://github.com/keshavgbpecdelhi/MakesMathEasy.git
 + ad3714b...0daf4bd factors -> factors (forced update)
Branch 'factors' set up to track remote branch 'factors' from 'origin'.

Keshav Raturi@DESKTOP-KEALHUA MINGW64 ~/Desktop/MakesMathEasy/MakesMathEasy (factors)
$

### PLS IGNORE THIS

sairish2001 reviewed (changes requested):
This PR is showing alot of Deletions. Have a look. Till then you can move on to any other issue, if you want to work on.

sairish2001 commented:
And also this issue is already resolved by someone
